From 54c4817fb91ff89540bc46fb2de1f72f170b4c5f Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.de>
Date: Tue, 2 Dec 2014 18:14:08 +0100
Subject: [PATCH 18/30] usb: hub: modify hub driver for missing ACPI parts
References: FATE#317453
Git-Commit: 54c4817fb91ff89540bc46fb2de1f72f170b4c5f
Patch-Mainline: Never (special patch needed to deal with less ACPI)

LPM uses information from the USB ACPI patches. We haven't taken
those. That part needs to be ripped out.

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
 drivers/usb/core/hub.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 23808a0..d4613a0 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -3739,16 +3739,10 @@ static int hub_set_address(struct usb_device *udev, int devnum)
  */
 static void hub_set_initial_usb2_lpm_policy(struct usb_device *udev)
 {
-	int connect_type;
-
 	if (!udev->usb2_hw_lpm_capable)
 		return;
 
-	connect_type = usb_get_hub_port_connect_type(udev->parent,
-			udev->portnum);
-
-	if ((udev->bos->ext_cap->bmAttributes & USB_BESL_SUPPORT) ||
-			connect_type == USB_PORT_CONNECT_TYPE_HARD_WIRED) {
+	if ((udev->bos->ext_cap->bmAttributes & USB_BESL_SUPPORT)) {
 		udev->usb2_hw_lpm_allowed = 1;
 		usb_set_usb2_hardware_lpm(udev, 1);
 	}
-- 
1.8.4.5

